File: CS/PrintCustomGroupRows/Converters.cs
using System;
using System.Globalization;
using System.Windows.Data;

namespace PrintCustomGroupRows {
    public class RemoveParenthesesConverter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            var str = value as string;
            if (str == null)
                return null;
            return str.Replace("(", string.Empty).Replace(")", string.Empty);
        }
        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) {
            throw new NotImplementedException();
        }
    }
    public class PrintRowInfoToImageSourceConverter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            Category category;
            if (Enum.TryParse(value.ToString(), out category)) {
                switch (category) {
                    case Category.Deferred:
                        return "Images/Deffered_32x32.png";
                    case Category.Normal:
                        return "Images/Normal_32x32.png";
                    case Category.Urgent:
                        return "Images/Urgent_32x32.png";
                }
            }
            return null;
        }
        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) {
            throw new NotImplementedException();
        }
    }
}

File: VB/PrintCustomGroupRows/Converters.vb
Imports System
Imports System.Globalization
Imports System.Windows.Data

Namespace PrintCustomGroupRows

    Public Class RemoveParenthesesConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim str = TryCast(value, String)
            If Equals(str, Nothing) Then Return Nothing
            Return str.Replace("(", String.Empty).Replace(")", String.Empty)
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class

    Public Class PrintRowInfoToImageSourceConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim category As Category
            If [Enum].TryParse(value.ToString(), category) Then
                Select Case category
                    Case Category.Deferred
                        Return "Images/Deffered_32x32.png"
                    Case Category.Normal
                        Return "Images/Normal_32x32.png"
                    Case Category.Urgent
                        Return "Images/Urgent_32x32.png"
                End Select
            End If

            Return Nothing
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace
